hail/src/main/scala/is/hail/linalg/LAPACK.scala

```diff
@@ -0,0 +1,87 @@
+package is.hail.linalg
+
+import java.lang.reflect.Method
+
+import com.sun.jna.{FunctionMapper, Library, Native, NativeLibrary, Pointer}
+import com.sun.jna.ptr.{DoubleByReference, IntByReference}
+
+import scala.util.{Failure, Success, Try}
+import is.hail.utils._
+
+class UnderscoreFunctionMapper extends FunctionMapper {
+  override def getFunctionName(library: NativeLibrary, method: Method): String = {
+    method.getName() + "_"
+  }
+}
+
+object TryAll {
+  def apply[K](f: => K): Try[K] =
+    try {
+      Success(f)
+    } catch {
+      case e: Throwable => Failure(e)
+    }
+}
+
+object LAPACK {
+  lazy val libraryInstance = {
+    val standard = Native.loadLibrary("lapack", classOf[LAPACKLibrary]).asInstanceOf[LAPACKLibrary]
+
+    versionTest(standard) match {
+      case Success(version) =>
+        log.info(s"Imported LAPACK version ${version} with standard names")
+        standard
+      case Failure(exception) =>
+        val underscoreAfterMap = new java.util.HashMap[String, FunctionMapper]()
+        underscoreAfterMap.put(Library.OPTION_FUNCTION_MAPPER, new UnderscoreFunctionMapper)
+        val underscoreAfter = Native.loadLibrary("lapack", classOf[LAPACKLibrary], underscoreAfterMap).asInstanceOf[LAPACKLibrary]
+        versionTest(underscoreAfter) match {
+          case Success(version) =>
+            log.info(s"Imported LAPACK version ${version} with underscore names")
+            underscoreAfter
+          case Failure(exception) =>
+            throw exception
+        }
+    }
```

Comment:
Scala is a funny language. I'm not sure I find this any easier to read than:
```scala
try {
  versionTest(standard)
  log.info(...)
  standard
} catch { case _: Exception =>
  val underscoreAfterMap = ...
  val underscoreAfter = ...
  versionTest(underscoreAfter)
  log.info(...)
  underscoreAfter
}
```
I suppose your implementation is clear about where an exception can occur. That's a good thing. My implementation puns, and has the unfortunate effect of also catching errors on the log line. It's also not strikingly obvious that the second `versionTest` will explode. I guess I prefer your implementation.